services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,168 @@
+  var cfenv = require('cfenv');
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then they are stored in sUsername and sPassword, as the 
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+  // Function that checks the payload and determines
+  // whether it is JSON or a Buffer
+  var checkPayload = function(payload) {
+    var message = null;
+    var isBuffer = false;
+
+    var hasJSONmethod = (typeof payload.toJSON === 'function') ;
+
+    if (hasJSONmethod === true) {
+      if (payload.toJSON().type === 'Buffer') {
+        isBuffer = true;
+      }      
+    }      
+    // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+    if (typeof payload !== 'string' &&  isBuffer !== true) {
+      message = 'The payload must be either a string or a Buffer';
+    }
+
+    return message;
+  }
+
+
+  // Function that checks the configuration to make sure that credentials,
+  // payload and options have been proviced in the correct format.
+  var checkConfiguration = function(msg, node, cb) {
+    var message = null;      
+    var taSettings = {};
+
+    // Credentials are needed for each of the modes.
+    username = sUsername || node.credentials.username;
+    password = sPassword || node.credentials.password;      
+
+    if (!username || !password) {
+      message = 'Missing Tone Analyzer service credentials';
+    }
+
+    if (!message && !msg.payload) {
+      message = 'Missing property: msg.payload';
+    } 
+
+    if (!message) {
+      message = checkPayload(msg.payload);
+    }
+
+    if (!message) {
+      taSettings.username = username;
+      taSettings.password = password;
+      //taSettings.tones = msg.tones || config.tones;
+      //taSettings.sentences = msg.sentences || config.sentences;
+      //taSettings.contentType = msg.contentType || config.contentType
+    }
+
+    if (cb) {
+      cb(message, taSettings);
+    }
+  };
+
+
+  // function when the node recieves input inside a flow. 
+  // Configuration is first checked before the service is invoked.
+  var processOnInput = function(msg, config, node) {
+    checkConfiguration (msg, node, function(err, settings){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'anonymous' has a complexity of 6.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2653056627/issues/source?bid=3225115&fileBranchId=3325196#l106)